src/python/pants/init/options_initializer.py

```diff
@@ -159,6 +154,19 @@ def compute_pantsd_invalidation_globs(buildroot, bootstrap_options):
                     f"Changes to {glob}, outside of the buildroot, will not be invalidated."
                 )
 
+        # Explicitly specified globs are already relative, and are added verbatim.
+        invalidation_globs.update(
+            (
+                "!*.pyc",
+                "!__pycache__/",
+                # TODO: This is a bandaid for https://github.com/pantsbuild/pants/issues/7022:
+                # macros should be adapted to allow this dependency to be automatically detected.
+                "requirements.txt",
+                "3rdparty/**/requirements.txt",
```

Comment:
It's a bit arbitrary, people may have requirements.txt files anywhere in their tree, but as long as we document how to modify this, it's OK.

Reply:
Yea, it is. But this felt like an ok bandaid that avoided walking the entire repository for pantsd to watch its invalidation globs (which it eagerly rechecks whenever they might be invalidated).

Reply:
This did, in fact, bite at least one person with a file at `src/requirements.txt`.